modules/social_features/social_profile/src/SocialProfileRenderService.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Drupal\social_profile;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\profile\Entity\ProfileInterface;
+use Drupal\user\Entity\User;
+use Drupal\user\UserInterface;
+
+/**
+ * Class SocialProfileRenderService.
+ *
+ * @package Drupal\social_profile
+ */
+class SocialProfileRenderService {
```

Comment:
We may want to add a way to invalidate cache entries here. We never know if calling code knows better than we do that a profile has changed between renders (even if the case is unlikely).